auth0/go-auth0
# Add Organizations for Client Credentials

### 🔧 Changes
1. Added new field called```default_organization```  to GET/PATCH endpoints ```/api/v2/clients/{id} ```
2. Added new field called```default_organization```  to POST endpoints  ```/api/v2/clients/```
3. Support query param ```q``` for GET ```/api/v2/clients```
      - q=client_grant.organization_id:org_sampleorgid213
      - q=client_grant.allow_any_organization:true
4. Support query param ```identifiers``` for GET ```/api/v2/organizations/{org_id}/client-grants```
      - identifiers=https%3A%2F%2Fmy-api.example.com
5. Support query param ```grant_ids``` for GET ```/api/v2/resource-servers```
      - grant_ids=cgr_123456789012,cgr_abcdefghijkl



### 📚 References

Required changes have been made in line with the [documentation here](https://oktawiki.atlassian.net/wiki/spaces/IAMPS/pages/3052704289/Organizations+for+Client+Credentials+GA+SDK+Changes)
[Management API docs here](https://sus.auth0.com/docs/api/management/v2/organizations/post-organizations)



### 🔬 Testing
Relevant test cases and their recordings have been added to ensure changes as described above.
All created resources are cleaned up after the assertions are completed.

```make test FILTER=TestOrganizationManager_ClientGrantsWithOrg```

- Create an Organization and ResourceServer
- Create a client with DefaultOrganization parameter and set the org.
- Create a clientGrant under the client created before.
- Associate the clientGrant with the Organization created before.
- List clients filtered with query param and assert.
- List clientGrants filtered with query param and assert.
- Remove the associated ClientGrants.
- Re-list clientGrants filtered with query param and ensure length is 0.
- Delete the clientGrant.
- Retrieve the clientGrant and ensure error is return since grant has been deleted.
- Delete all pending resources as part of the cleanup process.




### 📝 Checklist
